### PR TITLE
Add restore() that calls mockFn.mockRestore()

### PR DESCRIPTION
**Pull request by sergeysolovev**
About mockFn.mockRestore()
https://facebook.github.io/jest/docs/mock-function-api.html#mockfnmockrestore

This is not yet supported in Jest: https://github.com/facebook/jest/issues/2965

### CODE REVIEW

**Comment by sergeysolovev** [comment]
@chmanie Is it possible to get any feedback from you about the PR? This is my first PR, maybe I did something wrong.

File: __tests__/jest-sandbox.test.js [hunk 1]
```diff
@@ -66,3 +66,30 @@ test('Resets all mocks', () => {
...
+    .mockImplementation(() => 'yo, homie');
+  fn('a');
+  expect(fn).toBeCalledWith('a');
+  expect(fn.mock.calls).toHaveLength(1);
+  expect(fn.mock.instances).toHaveLength(1);
+  const cool = new CoolClass();
+  expect(cool.hello()).toBe('yo, homie');
+  expect(spy.mock.calls).toHaveLength(1);
+  expect(spy.mock.instances).toHaveLength(1);
+  sandbox.restore();
```
Comment (chmanie) [inline __tests__/jest-sandbox.test.js]: You're not testing the actual implementation here. We don't need the fn and what we actually want to test is the restore functionality. So we have to check if the implementation was overridden by the mockImplementation and then was restored after calling sandbox.restore

Comment (sergeysolovev) [inline __tests__/jest-sandbox.test.js]: @chmanie Thank you for your feedback! Yes, I can remove the lines 77, 82-84. Though, according to [the jest docs](https://facebook.github.io/jest/docs/en/mock-function-api.html#mockfnmockrestore):
> Beware that mockFn.mockRestore only works when mock was created with jest.spyOn. Thus you have to take care of restoration yourself when manually assigning jest.fn().

That's why I decided to test that it doesn't crash to call restore() when we have an `fn` in the sandbox.

The lines 87-88, 93-94 assert that `mockRestore` works as expected i.e. it [doesn't clear calls and instances](https://facebook.github.io/jest/docs/en/mock-function-api.html#mockfnmockclear):
> mockFn.mockClear()
Resets all information stored in the mockFn.mock.calls and mockFn.mock.instances arrays.

I can remove these lines either.

What do you think?

Comment (chmanie) [inline __tests__/jest-sandbox.test.js]: In general, we don't need to test jest functionality again (we just assume that it works and is tested itself). So we just have to test if the jest functions were called correctly.

Comment (sergeysolovev) [inline __tests__/jest-sandbox.test.js]: Sounds right to me, thank you for you support!

**Comment by chmanie** [comment]
All good now. Thanks again. I published version 1.1.0.

**Comment by sergeysolovev** [comment]
Good news, thanks!

**Pull request merged**